bin/runtime-common/src/refund_relayer_extension.rs

```diff
@@ -215,18 +190,23 @@ where
 
 	fn validate(
 		&self,
-		who: &Self::AccountId,
+		_who: &Self::AccountId,
 		call: &Self::Call,
-		info: &DispatchInfoOf<Self::Call>,
-		len: usize,
+		_info: &DispatchInfoOf<Self::Call>,
+		_len: usize,
 	) -> TransactionValidity {
-		// reject batch transactions with obsolete headers
-		if let Some(UtilityCall::<R>::batch_all { ref calls }) = call.is_sub_type() {
-			for nested_call in calls {
-				let reject_obsolete_transactions = BE::default();
-				reject_obsolete_transactions.pre_dispatch(who, nested_call, info, len)?;
-			}
+		let calls = match self.expand_call(call) {
```

Comment:
This has changed behavior a bit - we don't support rejecting obsolete stuff e.g. in `BatchAll { MessagesDelivery, MessagesDelivery }` anymore. This isn't critical for us - just caught my attention

Reply:
Oh, I didn't think of that. Changed it to do:
```
			call.check_obsolete_submit_finality_proof()?;
			call.check_obsolete_submit_parachain_heads()?;
			call.check_obsolete_receive_messages_proof()?;
```
for all nested calls in batches with at most 3 calls. 

But what would be the desired behavior here ? I see a couple of possibilities:
- checking for all possible obsolete headers and messages for any nested calls (as was done before with BE::validate()). But if in the future we'll support multiple `RefundRelayerForMessagesFromParachain` instances, I guess we don't want to check for all possible obsolete headers and messages in all the instances of `RefundRelayerForMessagesFromParachain`.  
- checking the batches supported by this extension (the batches contains at most 3 calls) with BE::validate()
- checking the batches supported by this extension (the batches contains at most 3 calls) with 
  ```
  call.check_obsolete_submit_finality_proof()?;
  call.check_obsolete_submit_parachain_heads()?;
  call.check_obsolete_receive_messages_proof()?;
  ```

Or maybe we can just extend `BridgeRejectObsoleteHeadersAndMessages` to support batch calls, which should cover everything. And remove this check from here.

Just trying to understand what the desire behavior is since this came up. But I would implement it in a future PR since it can get complicated.

Reply:
TBH - I don't know if we need it or not - just spotted the difference :)

> checking for all possible obsolete headers and messages for any nested calls (as was done before with BE::validate()). But if in the future we'll support multiple RefundRelayerForMessagesFromParachain instances, I guess we don't want to check for all possible obsolete headers and messages in all the instances of RefundRelayerForMessagesFromParachain.

IDK if we'll ever need more than one instance of this extension. Now the plan is to bridge two bridge hubs only and then use HTMP to pass messages between parachains. So maybe we don't need it at all. But in any case - iiuc we don't do any heavy ops there - its merely a `match call { ... }` and only if we find a bridge call, we do some actual checks (db reads and comparisons). So maybe it isn't a bog issue?

IMO let's leave it in current state for now - we may revisit it later.